requirements-dev.txt

```diff
@@ -1,72 +1,80 @@
 attrs==22.1.0
-black==22.8.0
+black==22.12.0
+cachetools==5.2.0
 certifi==2022.12.7
 cffi==1.15.1
+chardet==5.1.0
 charset-normalizer==2.1.1
 click==8.1.3
-coverage==6.4.4
+colorama==0.4.6
+coverage==6.5.0
 craft-cli==1.2.0
 craft-parts==1.17.1
 craft-providers==1.6.2
 craft-store==2.3.0
 cryptography==3.4
 Deprecated==1.2.13
 distlib==0.3.6
-filelock==3.8.0
-flake8==5.0.4
-humanize==4.3.0
-idna==3.3
+exceptiongroup==1.0.4
+filelock==3.8.2
+flake8==6.0.0
+humanize==4.4.0
+idna==3.4
+importlib-metadata==5.1.0
 iniconfig==1.1.1
+jaraco.classes==3.2.3
 jeepney==0.8.0
 Jinja2==3.1.2
-jsonschema==4.15.0
-keyring==23.8.2
+jsonschema==4.17.3
+keyring==23.11.0
 macaroonbakery==1.3.1
 MarkupSafe==2.1.1
 mccabe==0.7.0
+more-itertools==9.0.0
 mypy-extensions==0.4.3
-ops==1.5.2
-overrides==6.2.0
-packaging==21.3
-pathspec==0.10.0
-platformdirs==2.5.2
+ops==1.5.4
+overrides==7.3.1
+packaging==22.0
+pathspec==0.10.3
+platformdirs==2.6.0
 pluggy==1.0.0
-protobuf==3.20.2
-py==1.11.0
-pycodestyle==2.9.1
+protobuf==3.20.3
+pycodestyle==2.10.0
 pycparser==2.21
 pydantic==1.9.0
-pydantic-yaml==0.8.0
+pydantic-yaml==0.8.1
 pydocstyle==6.1.1
-pyflakes==2.5.0
+pyflakes==3.0.1
 pymacaroons==0.13.0
 PyNaCl==1.5.0
-pyparsing==3.0.9
+pyproject_api==1.2.1
 pyRFC3339==1.1
-pyrsistent==0.18.1
-pytest==7.1.2
+pyrsistent==0.19.2
+pytest==7.2.0
 pytest-mock==3.10.0
 python-dateutil==2.8.2
-pytz==2022.2.1
+pytz==2022.6
 pyxdg==0.28
 PyYAML==6.0
 requests==2.28.1
-requests-toolbelt==0.9.1
+requests-toolbelt==0.10.1
 requests-unixsocket==0.3.0
-responses==0.21.0
+responses==0.22.0
 SecretStorage==3.3.3
 semantic-version==2.10.0
 semver==2.13.0
-setuptools-rust==1.5.1
+setuptools-rust==1.5.2
 six==1.16.0
-snap-helpers==0.2.0
+snap-helpers==0.3.1
 snowballstemmer==2.2.0
-tabulate==0.8.10
+tabulate==0.9.0
 toml==0.10.2
 tomli==2.0.1
-tox==3.25.1
+tox==4.0.11
```

Comment:
Tox 4.0 has a fairly different configuration setup - I'm willing to bet it's what's causing the test failures. If we pin it <4.0 for the moment it should fix the breakages for now.